microservices-connector/cmd/router/main.go

```diff
@@ -726,8 +840,26 @@ func handleMultipartError(writer *multipart.Writer, err error) {
 
 func initializeRoutes() *http.ServeMux {
 	mux := http.NewServeMux()
-	mux.HandleFunc("/", mcGraphHandler)
-	mux.HandleFunc("/dataprep", mcDataHandler)
+
+	// Wrap connector handlers with otelhttp wrappers
+	// "http.server.request.size" -  Int64Counter -  "Measures the size of HTTP request messages" (Incoming request bytes total)
+	// "http.server.response.size" - Int64Counter  - "Measures the size of HTTP response messages" (Incoming response bytes total)
+	// "http.server.duration" - Float64histogram "Measures the duration of inbound HTTP requests." (Incoming end to end duration, milliseconds)
+	handleFunc := func(pattern string, handlerFunc func(http.ResponseWriter, *http.Request), operation string) {
+		handler := otelhttp.NewHandler(otelhttp.WithRouteTag(pattern, http.HandlerFunc(handlerFunc)), operation)
+		mux.Handle(pattern, handler)
+	}
+
+	handleFunc("/", mcGraphHandler, "mcGraphHandler")
+	handleFunc("/dataprep", mcDataHandler, "mcDataHandler")
+
+	// TODO: what is option as a string?!?!
+	//mux.Handle("/", otelhttp.NewHandler(http.HandlerFunc(mcGraphHandler), "mcGraphHandler"))
+	//mux.Handle("/dataprep", otelhttp.NewHandler(http.HandlerFunc(mcDataHandler), "mcDataHandler"))
+
+	// Export metric using Prometheus client built-in handler.
+	mux.Handle("/metrics", promhttp.Handler())
```

Comment:
Hi @ppalucki, is it possible to registry this router `metrics`  via function `handleFunc` as well?

Reply:
@zhlsunshine fixed
yes, it is possible, but didn't wrappe it on purpose to exclude from metrics 

but you're right - it should be covered and exlucded later in visualization/post processing